WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -412,6 +414,8 @@ public void onCreate(Bundle savedInstanceState) {
                                        .get(ReaderPostListViewModel.class);
         mNewsCardViewModel = ViewModelProviders.of(requireActivity(), mViewModelFactory)
                                                .get(NewsCardViewModel.class);
+        mReaderViewModel = ViewModelProviders.of(getParentFragment(), mViewModelFactory)
```

Comment:
The ReaderPostListFragment is used even on different screens than just the Reader entry point - eg. Open Reader -> Click on a blog title of one of the posts in the discovery tab -> notice ReaderPostListActivity + ReaderPostListFragment gets created. 
I believe the `getParentFragment` might return null as there isn't any parent fragment in this scenario. Even if the parent fragment existed it wouldn't be `ReaderFragment` and ReaderViewModel wouldn't exist so this would create a duplicate of `ReaderViewModel` which would never be used.
One solution is to wrap this into `mIsTopLevel` - it's a flag indicating we are in the top level (Reader entry point) screen. We can discuss other possible solution on a call/slack.

Wdyt?

Reply:
> We can discuss other possible solution on a call/slack.

That would be good because I lack the 'reader' background and your insight is greatly appreciated.